sphinx_gallery/gen_rst.py

```diff
@@ -231,6 +235,22 @@ def text2string(content):
         return content
 
 
+def extract_thumbnail_number(text):
+    """ Pull out the thumbnail image number specified in the docstring. """
+
+    # check whether the user has specified a specific thumbnail image
+    pattr = re.compile("^#\s*sphinx_gallery_thumbnail_number:\s*([0-9]+)\s*$", flags=re.MULTILINE)
+    match = pattr.search(text)
+    try:
+        thumbnail_number = int(match.groups()[0])
+    except AttributeError:
+        thumbnail_number = 1 # by default, use the first figure created
+
+    # remove the thumbnail configuration option from the document
+    text = pattr.sub("", text)
```

Comment:
I am not sure I would bother removing the thumbnail configuration from the example source code to be honest. For example I would very well imagine my self doing something like:

```python

# The next line is to pick the figure showing the negative exponential in orange
# sphinx_gallery_thumbnail_number = 2
```

in which case removing the second line is slightly confusing.